src/lua/zerobrane.lua

```diff
@@ -67,6 +77,114 @@ local function setZeroBranePreferences()
 	file = File(zbpreferencespath)
 
 	file:writeLine(table.concat(output, "\n"))
+	file:close()
+end
+
+local function setupPackageFile(fileName, previousFile, nextFile)
+	local file = File(fileName)
+	local line = file:readLine()
+	local fileBegin = line
+	local output = {}
+	table.insert(output, string.format("--[[ [previous](%s) | [contents](00-contents.lua) | [next](%s) ]]\n", previousFile, nextFile))
+	foundLicense = false
+	for _ = 1, licenseTextLines do
+		if not line then
+			break
+		end
+
+		if string.find(line, licenseTextPart) then
+			foundLicense = true
+		end
+
+		line = file:readLine()
+	end
+
+	if foundLicense then
+		line = file:readLine() -- after license
+	else
+		line = fileBegin -- back to begin to read all file
+	end
+
+	while line do
+		table.insert(output, line)
+		line = file:readLine()
+	end
+
+	file:close()
+	file = File(fileName)
+	local contents = table.concat(output, "\n")
+	file:writeLine(contents)
+	file:close()
+end
+
+local function createPackageIndex(packageName, packageExamplesPath, luaFiles, title)
+	local file = File(packageExamplesPath.."/00-contents.lua")
+	file:writeLine("--[[ [previous](00-contents.lua) | [back to index](../welcome.lua) | [next]("..luaFiles[1].indexed..")")
+	file:writeLine("\n# ".._Gtme.stringToLabel(title).." of ".._Gtme.stringToLabel(packageName).."\n")
+	for i = 1, #luaFiles do
+		file:writeLine(string.format("(%02d) [%s](%s)", i, string.gsub(luaFiles[i].fileName, ".lua", ""), luaFiles[i].indexed))
+	end
+
+	file:writeLine("]]")
+	file:close()
+end
+
+local function copyPackage(packageName, examplesPath, subfolders)
```

Comment:
It seems that `subfolders` is unnecessary. Use it as a local variable:
```lua
subfolders = {"examples", "data"}
```

It should also work for base package, because it does not have any lua file in the data directory.